ENH: Add Linux aarch64, ppc64le, and macOS arm64 builds

matthewfeickert opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->

* Enable builds for platforms: linux-aarch64, linux-ppc64le, osx-arm64.
* Bump build number.
* Add tests source files for `testthat` tests.
* Comment out running the `testthat` tests given ongoing issues with `v8` that affect test requirement `gt`.
   - c.f. https://github.com/conda-forge/r-v8-feedstock/issues/47
   - c.f. https://github.com/conda-forge/libv8-feedstock/pull/85
* Add `r-codetools` as a test requirement to avoid

```
code for methods in class Rcpp_... was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
```

warnings.

* Add docs URL.
* MNT: Re-rendered with conda-build 25.5.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.07.29.16.05.27

_

* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matthewfeickert committed:
ENH: Add Linux aarch64, ppc64le, and macOS arm64 builds

* Enable builds for platforms: linux-aarch64, linux-ppc64le, osx-arm64.
* Bump build number.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/16640169213](https://github.com/conda-forge/conda-forge-webservices/actions/runs/16640169213). Examine the logs at this URL for more detail.</sub>

matthewfeickert commented:
@conda-forge/r-gdalraster this is ready for review.

mfansler commented on recipe/meta.yaml:
```diff
@@ -67,7 +67,7 @@ test:
   commands:
     - $R -e "library('gdalraster')"  # [not win]
     - $R -e "testthat::test_package('gdalraster', stop_on_failure=TRUE)"  # [not win]
-    - "\"%R%\" -e \"library('gdalraster')\""  # [win] 
+    - "\"%R%\" -e \"library('gdalraster')\""  # [win]
     - "\"%R%\" -e \"testthat::test_package('gdalraster', stop_on_failure=TRUE)\""  # [win]
```
The testthat tests aren't running. Maybe we can clean this up in this PR as well?

```suggestion
    - r-codetools
  source_files:
    - tests/
  commands:
    - $R -e "library('gdalraster')"                                          # [not win]
    - $R -e "testthat::test_file('tests/testthat.R', stop_on_failure=TRUE)"  # [not win]
    - "\"%R%\" -e \"library('gdalraster')\""                                          # [win]
    - "\"%R%\" -e \"testthat::test_file('tests/testthat.R', stop_on_failure=TRUE)\""  # [win]
```

matthewfeickert replied:
Thanks @mfansler. I'll do this manually as when I try to apply your suggested changes GitHub is giving me a strange

```
Applying suggestions on deleted lines is currently not supported
```

which I've never seen before. :?

<img width="1414" height="1061" alt="image" src="https://github.com/user-attachments/assets/555d5ded-d539-4c4e-9fef-95e23fc75167" />

matthewfeickert replied:
```
> testthat::test_package('gdalraster', stop_on_failure=TRUE)
> 
> 
No installed testthat tests found for gdalraster
+ exit 0
```

Seems like a pretty bad design choice on `testthat` to have no files found return an exit code of `0`. ☹️

mfansler reviewed (approved):
Thanks - platform changes look good.

Just suggesting to fix the testing while we're here.

matthewfeickert committed:
MNT: Postpone testthat test evaluation given v8 issues.

* Add tests source files for testthat tests.
* Comment out running the testthat tests given ongoing issues with v8 that
  affect test requirement gt.

* Add r-codetools as a test requirement to avoid

code for methods in class Rcpp_... was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)

warnings.

* Add docs URL.

matthewfeickert commented:
> Just suggesting to fix the testing while we're here.

@mfansler As the tests are failing and this is outside the scope of my original PR I'm going to suggest that this get fixed elsewhere. I don't have time for anything more than a driveby helper PR. I'd also recommend that the v1 recipe format be allowed for R packages, as the testing speed difference is night and day, which matters when people are time limited, which is everyone.

matthewfeickert commented:
```
Encountered problems while solving:
  - package r-gt-1.0.0-r44hc72bb7e_0 requires r-juicyjuice >=0.1.0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
...
   ├─ r-gt [0.10.0|0.10.1|...|1.0.0] would require
   │  └─ r-base >=4.3,<4.4.0a0 * but there are no viable options
   │     ├─ r-base [4.1.3|4.2.3|4.3.0|4.3.1], which cannot be installed (as previously explained);
   │     ├─ r-base [4.1.3|4.2.3|4.3.1|4.3.2], which cannot be installed (as previously explained);
   │     ├─ r-base [4.1.3|4.2.3|4.3.3], which cannot be installed (as previously explained);
   │     ├─ r-base [4.2.3|4.3.2|4.3.3], which cannot be installed (as previously explained);
   │     ├─ r-base [4.2.3|4.3.3], which cannot be installed (as previously explained);
   │     └─ r-base 4.3.3 conflicts with any installable versions previously reported;
   ├─ r-gt [0.11.0|0.11.1|1.0.0] would require
   │  └─ r-juicyjuice >=0.1.0 * but there are no viable options
   │     ├─ r-juicyjuice 0.1.0 would require
   │     │  └─ r-base >=4.2,<4.3.0a0 * but there are no viable options
   │     │     ├─ r-base [4.0.5|4.1.3|...|4.2.3], which cannot be installed (as previously explained);
   │     │     ├─ r-base [4.1.3|4.2.3|4.3.0|4.3.1], which cannot be installed (as previously explained);
   │     │     ├─ r-base [4.1.3|4.2.3|4.3.1|4.3.2], which cannot be installed (as previously explained);
   │     │     ├─ r-base 4.2.3 conflicts with any installable versions previously reported;
   │     │     ├─ r-base [4.1.3|4.2.3|4.3.3], which cannot be installed (as previously explained);
   │     │     ├─ r-base [4.2.3|4.3.2|4.3.3], which cannot be installed (as previously explained);
   │     │     └─ r-base [4.2.3|4.3.3], which cannot be installed (as previously explained);
...
```

matthewfeickert commented:
Requires PR https://github.com/conda-forge/r-juicyjuice-feedstock/pull/4 to go in first.

mfansler commented:
> > Just suggesting to fix the testing while we're here.
> 
> @mfansler As the tests are failing and this is outside the scope of my original PR I'm going to suggest that this get fixed elsewhere.

Sounds good - feel free to revert and I'll merge it when it passes again.

matthewfeickert commented:
> Sounds good - feel free to revert and I'll merge it when it passes again.

Well, let me be a little bit less of a drama llama and see if I can get things working once `r-juicyjuice` gets rebuilt in https://github.com/conda-forge/r-juicyjuice-feedstock/pull/4.

mfansler commented:
Both `r-gt` and `r-juicyjuice` are `noarch` recipes so they are unlikely the root cause of the conflict. Rather, this is about `r-v8` lacking builds against newer `icu` (see https://github.com/conda-forge/r-v8-feedstock/issues/47). I'd say let's postpone getting the testthat tests working here until we address the `r-v8` problem.

PS: The root problem is https://github.com/conda-forge/libv8-feedstock/pull/85 if anyone wants to work on that.

mfansler commented on recipe/meta.yaml:
```diff
@@ -60,18 +60,22 @@ requirements:
     - r-nanoarrow
     - r-wk
     - r-xml2
+    - r-codetools
```
This shouldn't be in `run` - it's not necessary to install or run the package. One can put it in `host` and/or `test` if you'd like to avoid the warning messages during install and/or test time, resp.. I typically just do `test`.

```suggestion
```

matthewfeickert replied:
Okay, but what is the origin of the warning then? Getting multiple warnings means that there is a `gdalraster` issue (which should be addressed), or there is a dependency issue (which should get addressed), or the warnings are totally useless. 

It is very possible that I am missing something totally obvious, but a screenful of runtime warnings for anything that isn't TensorFlow indicates significant problems to me.

ctoney commented:
Sorry, I'm responsible for changing `testthat::testfile()` as @matthewfeickert had originally, to `testthat::test_package()` because the tests didn't appear to run with `test_file()`. I'm a little out of my element here but trying to get up to speed. I'll be glad to revisit the tests.

The tests in `gdalraster` that require `r-gt` and hence `r-v8` are a very minor part of the test suite that could be skipped conditionally with minimal consequence (e.g., `testthat::skip_if_not_installed("pkg")`), so that could be one option (`r-gt` is only a Suggested package for `gdalraster`, and is a heavy dependency, so I would view that as a helpful change in the test suite anyway). However, that would not make it into a CRAN release until `gdalraster` 2.2.0 around mid to late August. Considering only architecture differences, we do have fairly extensive testing across combined dev CI, CRAN and r-universe. Temporarily skipping tests here might be reasonable.

matthewfeickert committed:
MNT: Re-rendered with conda-build 25.5.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.07.29.16.05.27

matthewfeickert commented:
> I'd say let's postpone getting the testthat tests working here until we address the `r-v8` problem.

Thanks for all the very useful advice and for catching my mistakes on this, @mfansler. The context on the `v8` issues is definitely useful.

I've moved summary and discussion points to Issue https://github.com/conda-forge/r-gdalraster-feedstock/issues/2 for @ctoney to decide what they'd like to do for both running tests and addressing Windows builds.

ctoney reviewed (approved):
Thanks! I'll track down the issue with `codetools` warning.

mfansler reviewed (approved):
Thanks ❤️ , and apologies for the circuitous suggestion - wasn't expecting it be such a bother!

mfansler commented:
@ctoney no worries - you shouldn't have to do anything with `codetools` - that is an optional package that CRAN will by default have installed when it tests. On Conda Forge, we test with only `r-base`, but an end-user who wants a more default R installation from Conda Forge can use the `r` metapackage, which will include `r-recommended`, the "recommended" packages from CRAN among which is `r-codetools`.

matthewfeickert commented:
> On Conda Forge, we test with only `r-base`, but an end-user who wants a more default R installation from Conda Forge can use the `r` metapackage, which will include `r-recommended`, the "recommended" packages from CRAN among which is `r-codetools`.

I didn't know any of this (I'd only ever used `r-base`), so thanks very much for sharing! That's super useful.

matthewfeickert commented:
> Thanks ❤️ , and apologies for the circuitous suggestion - wasn't expecting it be such a bother!

It flushed out some important revisions, so it was useful, not bothersome. :+1:

ctoney commented:
@mfansler, yes thanks for the explanation on `codetools`. I hoped it was somewhere along those lines. Appreciate the information.
